frontend/src/components/card/Card.tsx

```diff
@@ -3,7 +3,7 @@ import classes from "./Card.module.scss";
 
 type CardProps = HTMLAttributes<HTMLDivElement> & {
   className?: string;
-  variant?: "blue" | "black";
+  variant?: "table-line" | "black" | "header-row" | "dark-blue";
```

Comment:
What happens if there are even mode variants added in the future? I think these can become harder to maintain and adapt for all the different style variants. 

If some components starts to adapt to many different (an possibly unrelated) use-cases it's usually better to generalise that component, so that it can be customised from the parent component itslef.

In this example, we could replace the `variant` prop with `className` prop, which would allow more extensibility from the parent, because consumers of this component won't be limited to the predefined "style variants", but could also implement their own "variant" by providing a custom class name.

Reply:
This is also related to a general concept of "composition" - https://reactjs.org/docs/composition-vs-inheritance.html.

Usually the "better way" is to make components easily composable and extendable - even to undefined future use-cases.